### PR TITLE
feat:adding release drafter configuration

### LINKED ISSUE #2359: Addition of Automated Release Notes:  ReleaseDrafter
**Is your feature request related to a problem or area of OpenRefine? Please describe.**

Currently, we are following manual releasing with help of ReleaseDrafter we can easily list out our changes in different sub-sections like: Chore, Feature, Enhancement, Bug etc.

**Describe the solution you'd like**

Addition of ReleaseDrafter, more details over here https://github.com/marketplace/actions/release-drafter

### PR DESCRIPTION
**Pull request by kushthedude**
Signed-off-by: kushthedude <kushthedude@gmail.com>

Fixes #2359

### CODE REVIEW

File: .github/release-drafter.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name-template: v$NEXT_PATCH_VERSION 🌈
+tag-template: v$NEXT_PATCH_VERSION
+categories:
+  - title: 🚀 Features
+    label: feature
+  - title: 🐛 Bug Fixes
+    label: fix
```
Comment (wetneb) [inline .github/release-drafter.yml]: This list would need to be adapted to the tags that we actually use: `enhancement`, `bug`, and so on. I would probably remove the icons - I don't think they add much.

**Comment by kushthedude** [comment]
@wetneb @thadguidry I have renamed and made the config file on the basis of the labels currently getting used in Open Refine.
Now to really take a benefit of the ReleaseDrafter we need to label the PRs accordingly. To facilitate the labelling better : We can have [Semantic PR Naming](https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716). Once we will have the semantic PR naming, I can make a GitHub bot in a day or two which may label the PR according to its name prefix. With this our complete release drafting process will get automated.
Please tell me how does this sound to you !

**Comment by thadguidry** [comment]
I have my own preferences, but @wetneb has been responsible for our last several releases, so I will let him make all the judgement calls for this.  Work with him directly and we should be in great shape.

**Comment by wetneb** [comment]
> Now to really take a benefit of the ReleaseDrafter we need to label the PRs accordingly. To facilitate the labelling better : We can have Semantic PR Naming. Once we will have the semantic PR naming, I can make a GitHub bot in a day or two which may label the PR according to its name prefix. With this our complete release drafting process will get automated.

We are already tagging issues, so in theory it should be possible to use these tags without having to duplicate them on the PRs themselves, assuming the PRs correctly use keywords to close issues, no?

**Comment by kushthedude** [comment]
> > Now to really take a benefit of the ReleaseDrafter we need to label the PRs accordingly. To facilitate the labelling better : We can have Semantic PR Naming. Once we will have the semantic PR naming, I can make a GitHub bot in a day or two which may label the PR according to its name prefix. With this our complete release drafting process will get automated.
> 
> We are already tagging issues, so in theory it should be possible to use these tags without having to duplicate them on the PRs themselves, assuming the PRs correctly use keywords to close issues, no?
> 
> If we can avoid writing our own custom bot for this, it would probably be better (since that avoids maintaining that code).

The releaser drafts the notes on the basis of the pull request since pull request title better signifies the gist of issue getting solved.

**Comment by wetneb** [comment]
Sure, that is obviously fine.

I would be happy to label pull requests too. What I mean is that it should be possible to have a GitHub action which automatically labels pull requests by copying the labels from the issue that the pull request closes, no?

**Comment by wetneb** [comment]
Closing this as stale, no activity for nearly a year. Feel free to reopen once the problem mentioned above is addressed (our PRs are not currently tagged, issues are, so the workflow would not work well as such).

**Pull request closed without merging**